### ISSUE TITLE
MUC Light - Changing roster item name field for a MUC light room

### ISSUE THREAD

**Issue by apparition47** [esl/MongooseIM#844]
MongooseIM version: master b13807d Merge pull request #577 from esl/muc_light
Installed from: source
Erlang/OTP version: 18

Steps:  Create room (room added to roster), set MUC light roomname by config, then get roster. However, when I try to update the "name" field of the room in my roster, it creates a duplicate JID entry instead of replacing it (works fine for non-MUC roster items). Any ideas? I'm not sure if this is related to #839.

```
XMPP OUT:
<iq from='crone1@shakespeare.lit/desktop'
    id='create1'
    to='coven@muclight.shakespeare.lit'
    type='set'>
        <query xmlns='urn:xmpp:muclight:0#create'>
    </query>
</iq>

<iq type='set' id='f84a58ed-fcc5-4dd2-9583-607af59333ba:roster' xmlns='jabber:client'><query xmlns='jabber:iq:roster'/><item jid="coven@muclight.shakespeare.lit" name="NEW NAME HERE"></query></iq>

<iq type='get' id='f84a58ed-fcc5-4dd2-9583-607af59333ba:roster' xmlns='jabber:client'><query xmlns='jabber:iq:roster'/></iq>
```

```
XMPP IN:
<iq from='crone1@shakespeare.lit' to='crone1@shakespeare.lit/desktop' id='f84a58ed-fcc5-4dd2-9583-607af59333ba:roster' type='result' xmlns='jabber:client'>
<query xmlns='jabber:iq:roster'>

<!-- HERE  two items with the same JID here -->
<item ask='subscribe' subscription='none' name='NEW NAME HERE' jid='coven@muclight.shakespeare.lit'><group>urn:xmpp:muclight:0</group></item>

<item subscription='to' name='Untitled' jid='coven@muclight.shakespeare.lit'><group>urn:xmpp:muclight:0</group><version>1464-815931-920455</version></item>

</query>
</iq>
```

**Comment by fenek** [comment]
It's not related to #839. MUC Light entries are not merged with user's entries.
Indeed it should be possible to set custom name for a room in a roster. It's an oversight on my side. :(

**Comment by apparition47** [comment]
Thanks @fenek . Also having trouble deleting any MUC group "contacts" from the XMPP roster. I send a `<iq>` to delete by JID but the group still remains on the roster when I go to retrieve it again. I'm guessing it's related to the above.

**Comment by apparition47** [comment]
Seems to be fixed with 2.0.0 beta 1. Thanks.
